### ISSUE TITLE
Failed to parse warning

### ISSUE THREAD

**Issue by jeroenbourgois** [jeremyjh/dialyxir#337]
### Environment

* Elixir & Erlang/OTP versions (elixir --version):
Erlang/OTP 22 [erts-10.4.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]

Elixir 1.8.2 (compiled with Erlang/OTP 21)

* Which version of Dialyxir are you using? (cat mix.lock | grep dialyxir):
` "dialyxir": {:hex, :dialyxir, "1.0.0-rc.6", "78e97d9c0ff1b5521dd68041193891aebebce52fc3b93463c0a6806874557d7d", [:mix], [{:erlex, "~> 0.2.1", [hex: :erlex, optional: false]}]},`

```
Failed to parse warning:
[{:"(", 1}, {:"#", 1}, {:"{", 1}, {:atom_full, 1, '\'__meta__\''}, {:"=>", 1}, {:"#", 1}, {:"{", 1}, {:atom_full, 1, '\'__struct__\''}, {:"=>", 1}, {:atom_full, 1, '\'Elixir.Ecto.Schema.Metadata\''}, {:",", 1}, {:atom_full, 1, '\'context\''}, {:"=>", 1}, {:atom_full, 1, '\'nil\''}, {:",", 1}, {:atom_full, 1, '\'prefix\''}, {:"=>", 1}, {:atom_full, 1, '\'nil\''}, {:",", 1}, {:atom_full, 1, '\'schema\''}, {:"=>", 1}, {:atom_full, 1, '\'Elixir.Cream.UserRole\''}, {:",", 1}, {:atom_full, 1, '\'source\''}, {:"=>", 1}, {:"<<", 1}, {:int, 1, 114}, {:",", 1}, {:int, 1, 111}, {:",", 1}, {:int, 1, 108}, {:",", 1}, {:int, 1, 101}, {:",", 1}, {:int, 1, 115}, {:",", 1}, {:int, 1, 95}, {:",", 1}, {:int, 1, 117}, {:",", 1}, {:int, 1, 115}, {:",", 1}, {:int, 1, 101}, {:",", 1}, {:int, 1, 114}, {:",", 1}, {:int, 1, 115}, {:">>", 1}, {:",", ...}, {...}, ...]


Legacy warning:
lib/cream/seeder.ex:83: The call 'Elixir.Cream.UserRole':changeset
         (#{'__meta__' =>
                #{'__struct__' => 'Elixir.Ecto.Schema.Metadata',
                  context => nil,
                  prefix => nil,
                  schema => 'Elixir.Cream.UserRole',
                  source =>
                      <<114,111,108,101,115,95,117,115,101,114,115>>,
                  state => built},
            '__struct__' => 'Elixir.Cream.UserRole',
            id => nil,
            inserted_at => nil,
            role =>
                #{'__cardinality__' => one,
                  '__field__' => role,
                  '__owner__' => 'Elixir.Cream.UserRole',
                  '__struct__' => 'Elixir.Ecto.Association.NotLoaded'},
            role_id => nil,
            updated_at => nil,
            user =>
                #{'__cardinality__' => one,
                  '__field__' => user,
                  '__owner__' => 'Elixir.Cream.UserRole',
                  '__struct__' => 'Elixir.Ecto.Association.NotLoaded'},
            user_id => nil},
          #{}) breaks the contract 
          ('Elixir.Cream.UserRole':t(), map()) ->
             'Elixir.Ecto.Changeset':t()
```

**Comment by asummers** [comment]
Seems it doesn't like the `__meta__` part for some reason.

**Comment by asummers** [comment]
Should be fixed in Erlex 0.2.2
